src/app/modules/shared_models/contacts_model.nim

```diff
@@ -86,6 +92,10 @@ QtObject:
     self.countChanged()
 
   proc addItem*(self: Model, item: Item) =
+    let ind = self.findIndexByPubKey(item.pubKey)
+    if(ind > -1):
```

Comment:
@jrainville if the point of this check is to add item only in case it was not added before, then that should be a case for a business logic part... and there is already a method `containsItemWithPubKey` which returns `true` if an item with a public key is already in the model. So please revert changes done here and update `onContactDetailsUpdated` method in the `module.nim`, it should look something like this:
```
method onContactDetailsUpdated*(self: Module, publicKey: string) =
  let contact = self.controller.getContactById(publicKey)
  if(contact.requestReceived() and 
    not contact.isContact() and 
    not contact.isBlocked() and
    not self.view.contactRequestsModel().containsItemWithPubKey(publicKey)):
      let item = self.createItemFromPublicKey(publicKey)
      self.view.contactRequestsModel().addItem(item)

  let (chatName, chatImage, isIdenticon) = self.controller.getOneToOneChatNameAndImage(publicKey)
  self.view.chatsModel().updateItemDetails(publicKey, chatName, chatImage, isIdenticon)
```